### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
## Summary

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

## Rationale

Typos enable bugs

## Testing Environment

The misspellings have been reported at https://github.com/jsoref/inspircd/commit/eb73fbc723c20780a20e9fe0c07f1a9dcac4f9e6#commitcomment-52328394

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/inspircd/commit/60b811e3520bb3ef8fc33fb59eb62b45d5b943ba

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I have tested this pull request on:

**Operating system name and version:** 
**Compiler name and version:** 

## Checks

I have ensured that:

  - [ ] This pull request does not introduce any incompatible API changes.
  - [ ] If ABI changes have been made I have incremented MODULE_ABI in `moduledefs.h`.
  - [ ] I have documented any features added by this pull request.

### CODE REVIEW

File: docs/conf/quotes.txt.example [hunk 1]
```diff
@@ -49,7 +49,7 @@ Computers make very fast, very accurate mistakes
 Life would be so much easier if we only had the source code
 Who is this 'General Failure' and why is he reading my disk?
 hAS aNYONE sEEN MY cAPSLOCK kEY?
-InspIRCd, now with excessive ammounts of Cheeze
+InspIRCd, now with excessive amounts of Cheeze
```
Comment (jsoref) [inline docs/conf/quotes.txt.example:52]: I'm leaving `Cheeze` -- happy to drop this, note that the spell-checker was told to ignore this file after the original list of typos was selected.

File: include/clientprotocolmsg.h [hunk 2]
```diff
@@ -407,17 +407,17 @@ struct ClientProtocol::Messages::Topic : public ClientProtocol::Message
  */
 class ClientProtocol::Messages::Privmsg : public ClientProtocol::Message
 {
-	void PushTargetChan(char status, const Channel* targetchan)
+	void PushTargetChan(char status, const Channel* targetchannel)
```
Comment (jsoref) [inline include/clientprotocolmsg.h:410]: Happy to drop this, both spellings were used.

File: include/serialize.h [hunk 3]
```diff
@@ -60,7 +60,7 @@ class CoreExport Serializable
 
 		/** Loads the value with the specified key.
 		 * @param key The key by which this data is identified.
-		 * @param out The location to store the value for this keu
+		 * @param out The location to store the value for this key.
```
Comment (jsoref) [inline include/serialize.h:63]: This matches the item above

File: make/template/inspircd [hunk 4]
```diff
@@ -395,7 +395,7 @@ sub getpidfile
 		$i =~ s/[^=]+=\s(.*)/\1/;
 		if (($i =~ s/\<include file=\"(.+?)\"\>//i) && ($i !~ /^#/))
 		{
-			# Decend into that file, and check for PIDs.. (that sounds like an STD ;/)
+			# Descend into that file, and check for PIDs.. (that sounds like an STD ;/)
```
Comment (jsoref) [inline make/template/inspircd:398]: Probably

File: src/modules.cpp [hunk 5]
```diff
@@ -55,7 +55,7 @@ Version::Version(const std::string &desc, int flags, const std::string& linkdata
 {
 }
 
-// These declarations define the behavours of the base class Module (which does nothing at all)
+// These declarations define the behaviours of the base class Module (which does nothing at all)
```
Comment (jsoref) [inline src/modules.cpp:58]: fwiw, this is en-GB

File: src/modules/extra/m_ssl_openssl.cpp [hunk 6]
```diff
@@ -336,7 +336,7 @@ namespace OpenSSL
 		/** OpenSSL makes us have two contexts, one for servers and one for clients
 		 */
 		Context ctx;
-		Context clictx;
+		Context clientctx;
```
Comment (jsoref) [inline src/modules/extra/m_ssl_openssl.cpp:339]: both spellings were used.

Note that I'm not picky / do tend to favor the spelling with more uses. Sometimes this means longer, sometimes shorter. Dropping isn't a big deal.

File: src/modules/m_censor.cpp [hunk 7]
```diff
@@ -55,8 +55,8 @@ class ModuleCensor : public Module
 		{
 			case MessageTarget::TYPE_USER:
 			{
-				User* targuser = target.Get<User>();
-				if (!targuser->IsModeSet(cu))
+				User* targetuser = target.Get<User>();
```
Comment (jsoref) [inline src/modules/m_censor.cpp:58]: Similar to targetchannel

File: src/modules/m_disable.cpp [hunk 8]
```diff
@@ -116,7 +116,7 @@ class ModuleDisable : public Module
 		commands.swap(newcommands);
 
 		// Whether we should fake the non-existence of disabled things.
-		fakenonexistent = tag->getBool("fakenonexistent", tag->getBool("fakenonexistant"));
+		fakenonexistent = tag->getBool("fakenonexistent", tag->getBool("fakenonexistent"));
```
Comment (jsoref) [inline src/modules/m_disable.cpp:119]: ok, this is wrong. What is this code trying to do?

File: src/modules/m_filter.cpp [hunk 9]
```diff
@@ -552,7 +552,7 @@ ModResult ModuleFilter::OnPreCommand(std::string& command, CommandBase::Params&
 		parameters[parting ? 1 : 0] = "Reason filtered";
 
 		/* We're warning or blocking, OR they're quitting and its a KILL action
-		 * (we cant kill someone whos already quitting, so filter them anyway)
+		 * (we cant kill someone who's already quitting, so filter them anyway)
```
Comment (jsoref) [inline src/modules/m_filter.cpp:555]: Most of the `whos` were `whose` this doesn't appear to be.

File: src/modules/m_messageflood.cpp [hunk 10]
```diff
@@ -154,7 +154,7 @@ class ModuleMsgFlood
 		{
 			if (f->addmessage(user, weight))
 			{
-				/* Youre outttta here! */
+				/* You're outttta here! */
```
Comment (jsoref) [inline src/modules/m_messageflood.cpp:157]: I left the cute `outttta`, but I don't think it needs `Youre`

File: win/NSIS.template.in [hunk 11]
```diff
@@ -489,15 +489,15 @@ Done:
 	Exch $R1
 FunctionEnd
 
-Function ConditionalAddToRegisty
+Function ConditionalAddToRegistry
```
Comment (jsoref) [inline win/NSIS.template.in:492]: A couple files have Windows line endings...

File: src/modules/m_disable.cpp (hunk 8 above)
Comment (genius3000) [reply src/modules/m_disable.cpp:119]: This is a previous typo fix for a config variable, so we still accept the former (incorrect) spelling to not break configs.

File: include/clientprotocolmsg.h (hunk 2 above)
Comment (SadieCat) [reply include/clientprotocolmsg.h:410]: `chan` is a common abbreviation of channel so it's probably fine to leave it.

File: docs/conf/quotes.txt.example (hunk 1 above)
Comment (DrummingAway) [reply docs/conf/quotes.txt.example:52]: thank you ;)

File: include/clientprotocolmsg.h (hunk 2 above)
Comment (DrummingAway) [reply include/clientprotocolmsg.h:410]: **great patch ;)**

**Review by genius3000 — APPROVED** [review]
LGTM
I'm impartial to the `targetchan` vs. `targetchannel` as I'm sure I even use both from time to time (as well as nick vs. nickname, etc.); as long as it's consistent within a module or section, it's all good. I do like the idea of `targ*` to `target*` though.

**Comment by SadieCat** [comment]
Can you revert the chan->channel patch? It's a common abbreviation which is used in several places. Other than that this LGTM.

**Comment by SadieCat** [comment]
Merged, thanks for your contribution! 👍

**Pull request merged**